From 212309075d26668f6e25f30754e646952843cf61 Mon Sep 17 00:00:00 2001
From: "Anthony G. Basile" <blueness@gentoo.org>
Date: Tue, 28 Apr 2015 18:56:46 -0600
Subject: [PATCH] 

On a uClibc system, the macro definitions of abs() and fabs()
 cause the build to fail.  Since these are specified by POSIX, it is safe to
 just drop their definition.  This has been tested on uClibc, glibc and musl.

See: https://bugs.gentoo.org/show_bug.cgi?id=510766

Signed-off-by: Anthony G. Basile <blueness@gentoo.org>
---
 server/include/misc.h | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/server/include/misc.h b/server/include/misc.h
index 2267ae0..fb904c8 100644
--- a/server/include/misc.h
+++ b/server/include/misc.h
@@ -71,12 +71,6 @@ typedef int Bool;
 
 #define min(a, b) (((a) < (b)) ? (a) : (b))
 #define max(a, b) (((a) > (b)) ? (a) : (b))
-#ifndef abs
-#define abs(a) ((a) > 0 ? (a) : -(a))
-#endif
-#ifndef fabs
-#define fabs(a) ((a) > 0.0 ? (a) : -(a))        /* floating absolute value */
-#endif
 #define sign(x) ((x) < 0 ? -1 : ((x) > 0 ? 1 : 0))
 /* this assumes b > 0 */
 #define modulus(a, b, d)    if (((d) = (a) % (b)) < 0) (d) += (b)
-- 
2.0.5

